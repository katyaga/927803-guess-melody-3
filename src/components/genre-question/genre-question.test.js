import React from 'react';
import renderer from 'react-test-renderer';
import GenreQuestion from './genre-question.jsx';

const answer = {
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  genre: `rock`,
};

const answers = {
  answers: [true, false]
};

const i = 2;

it(`GenreQuestion is rendered correctly`, () => {
  const tree = renderer.create((
    <GenreQuestion
      i={i}
      isPlaying={true}
      answers={answers}
      answer={answer}
      onChangeInput={() => {}}
      onPlayButtonClick={() => {}}
    />
  ), {
    createNodeMock: () => {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
