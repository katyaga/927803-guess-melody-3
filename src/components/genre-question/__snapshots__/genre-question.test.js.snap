// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestion is rendered correctly 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    type="button"
  />
  <div
    className="track__status"
  >
    <audio
      src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
    />
  </div>
  <div
    className="game__answer"
  >
    <input
      className="game__input visually-hidden"
      id="answer-2"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="answer-2"
    />
    <label
      className="game__check"
      htmlFor="answer-2"
    >
      Отметить
    </label>
  </div>
</div>
`;
